### PR TITLE
add 'kubectl kots gitops-info -n ns' to get gitops providers and keypairs

### PR DESCRIPTION
**Pull request by laverya**
this is a hidden command

This also moves the 'gitops' package into kots from kotsadm in order to be accessible to the CLI

### CODE REVIEW

File: pkg/gitops/gitops.go [hunk 1]
```diff
@@ -677,6 +683,75 @@ func CreateGitOpsCommit(gitOpsConfig *GitOpsConfig, appSlug string, appName stri
...
+
+		newIndex, err := strconv.ParseInt(splitKey[1], 10, 64)
+		if err != nil {
+			continue
+		}
+		indexes[newIndex] = struct{}{}
+	}
+
+	encryptionKeySecret, err := clientset.CoreV1().Secrets(namespace).Get(context.TODO(), "kotsadm-encryption", metav1.GetOptions{})
+	if kuberneteserrors.IsNotFound(err) {
```
Comment (salahalsaleh) [inline pkg/gitops/gitops.go:721]: i think this should fail if the encryption key is not found

**Comment by salahalsaleh** [comment]
why move `pkg/gitops/clone.go`, `pkg/gitops/key.go` and `pkg/gitops/types/gitops_interface.go`? i think they should stay in kotsadm as the functions/types defined there are all related to kotsadm functionality, not kots.

File: cmd/kots/cli/gitops-info.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			v := viper.GetViper()
+			log := logger.NewLogger()
+
+			namespace := v.GetString("namespace")
+			if namespace == "" {
+				log.Info("A namespace is required")
+				os.Exit(1)
+			}
+
+			keypairs, err := gitops.GetGitopsKeypairs(namespace)
```
Comment (salahalsaleh) [inline cmd/kots/cli/gitops-info.go:33]: i think this should be an API call instead of moving the kotsadm packages to kots

Comment (salahalsaleh) [inline cmd/kots/cli/gitops-info.go:33]: i think this should be an API call instead of moving the kotsadm packages to kots

**Review by salahalsaleh — CHANGES_REQUESTED** [review]
i think this should be an API call instead of moving the kotsadm packages to kots

**Comment by AmberAlston** [comment]
@laverya should this PR still be open?

**Pull request closed without merging**